INSTALL.md: some tweaks

fingolfin opened the pull request:
This is based on some recent Slack discussion. Should be backported, I think.

alex-konovalov commented on INSTALL.md:
```diff
@@ -44,8 +44,8 @@ operating system. In particular, it does not include an automatic installer
 Installing the GAP distribution with all the packages and full data libraries
-takes about 1.6 GB of disk space and (except on Windows) will require both a C
-and a C++ compiler (gcc is recommended) to be installed on your system.
+and (except on Windows) will require both a C and a C++ compiler (gcc or clang
```
check this sentence

wucas commented:
Sorry for the long comment, none of it means the PR cannot be merged. 
Regarding the command line option with which one apparently is supposed to start gap when running these tests I just noticed that this is even shown when I do start GAP with these options which might be confusing. Or at least I just double checked the options I used because I interpreted this as that I did not use these options. But that might just be me reading this wrong.

fingolfin commented:
Many good comments, I'll address them later. On describing the output of `Test` resp. `TestDirectory`: Yes, that should be done, but not in the README but rather in the documentation for those functions (and then the README can point to that).

wilfwilson commented on INSTALL.md:
```diff
@@ -44,8 +44,8 @@ operating system. In particular, it does not include an automatic installer
-takes about 1.6 GB of disk space and (except on Windows) will require both a C
-and a C++ compiler (gcc is recommended) to be installed on your system.
+and (except on Windows) will require both a C and a C++ compiler (gcc or clang
+are recommended) to be installed on your system.
```
I feel pretty sure that this should still be 
```suggestion
is recommended) to be installed on your system.
```
Of course it might be my peculiar interpretation of the English language, but with an 'or' statement, I like to use the same verb conjugation that I would use if I 'expanded' the statement:
> gcc or clang is recommend → gcc is recommended or clang is recommended

I don't mind if you disagree and want to keep the sentence as is. Otherwise, no objections to this PR 🙂

fingolfin commented:
Thanks everybody for the feedback. I tried to fix the language issues pointed out by @alex-konovalov and @wilfwilson, please have a look. I also updated another TestDirectory output example inside the reference manual to match what is in  `INSTALL.md` (and fixed the title of this Pr and the commit to reference `INSTALL.md`, not `README.md`... ooops). I also took into account what @wucas wrote, and (a) created issue #3776, (b) added a commit which removes the messages in the tests that recommends users to start GAP with certain options - I *never* heed those anyway, and I agree they are confusing. For people resp. occasions where the might matter (e.g. could be in CI builds), we can still set those options, of course.

fingolfin commented:
Backported to stable-4.11 in 7ae90d6f1
